context/llpcCompiler.cpp

```diff
@@ -1489,6 +1489,7 @@ MetroHash::Hash Compiler::GenerateHashForCompileOptions(
         cl::LogFileOuts.ArgStr,
         cl::EnableShadowDescriptorTable.ArgStr,
         cl::ShadowDescTablePtrHigh.ArgStr,
+        "amdgpu-scratch-bounds-checking",
```

Comment:
I don't think we should ignore option "amdgpu-scratch-bounds-checking" when generate hash code, because it will affect final GPU ISA code. I think we should enable this option in general and check the gfxip within LLVM back-end.

another choice is implement amdgpu-scratch-bounds-checking as an target option, then we can choose it in run time, instead of initial time.

Reply:
The LLVM backend code is target agnostic, it can be applied to any GFX revision.  So I don't think that's the correct place for the option.

A target option might work, I will investigate this.
However I wonder if another solution is to relax the check in ICompiler::Create to only compare option hashes for compilers of the same gfxIp?
I notice that GenerateHashForCompileOptions is only used in this context, so the question is whether the intention is to have compilers that produce identical output (which they won't for different GFX revision) or simply compatible output.

Reply:
- This hash is used in shader cache. so we can relex the check for same gfxip.
- but I am afraid we can't set different value for same option.

Reply:
I can see the problem in detail now
I hadn't appreciated the static reuse of the compiler options between all compiler instances.
I think this issue will need to be fixed by further changes to the feature implementation in the backend and frontend.
I'll close this merge request.